examples/basic_operations/add_responsive_search_ad.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This example adds a responsive search ad to a given ad group.
+
+To get ad groups, run get_ad_groups.py.
+"""
+
+
+import argparse
+import sys
+from uuid import uuid4
+
+from google.ads.google_ads.client import GoogleAdsClient
+from google.ads.google_ads.errors import GoogleAdsException
+
+
+def main(client, customer_id, ad_group_id):
+    ad_group_ad_service = client.get_service('AdGroupAdService', version='v2')
+    ad_group_service = client.get_service('AdGroupService', version='v2')
+
+    pinned_headline = client.get_type('AdTextAsset', version='v2')
+    pinned_headline.text.value = f'Cruise to Mars #{str(uuid4())[:8]}'
+    # Set a pinning to always choose this asset for HEADLINE_1. Pinning is
+    # optional; if no pinning is set, then headlines and descriptions will be
+    # rotated and the ones that perform best will be used more often.
+    pinned_headline.pinned_field = client.get_type(
+       'ServedAssetFieldTypeEnum', version='v2').HEADLINE_1
```

Comment:
Can't you make this be created by _create_ad_text_asset?
You probably can add one more param to choose whether to make the created text asset pinned or not.
This will head readability too because it will be closer to where it's used.

Reply:
fixed.